src/tls/openssl.c

```diff
@@ -312,6 +318,101 @@ static int windows_load_system_certificates(struct tls_context *ctx)
 }
 #endif
 
+#ifdef __APPLE__
+/* macOS-specific system certificate loading */
+static int macos_load_system_certificates(struct tls_context *ctx)
+{
+    X509_STORE *store = NULL;
+    X509 *x509 = NULL;
+    SecCertificateRef cert = NULL;
+    CFArrayRef certs = NULL;
+    CFDataRef certData = NULL;
+    const unsigned char *data = NULL;
+    char *subject = NULL;
+    char *issuer = NULL;
+    OSStatus status;
+    unsigned long err;
+    int ret = -1;
+    int loaded_cert_count = 0;
+
+    /* Retrieve system certificates from macOS Keychain */
+    status = SecTrustSettingsCopyCertificates(kSecTrustSettingsDomainSystem, &certs);
+    if (status != errSecSuccess || !certs) {
+        flb_debug("[tls] failed to load system certificates from keychain, status: %d", status);
+        return -1;
+    }
+
+    flb_debug("[tls] attempting to load macOS keychain system certificates");
```

Comment:
Maybe, we have to use: 

* [tls] attempting to load certificates in macOS's keychain system 
or 
* [tls] attempting to load certificates in system keychain of macOS
or
* [tls] attempting to load certificates in the system keychain

I sometimes similar writing way but, in English, continuing three or more nouns couldn't be clear what they mean.
Plus, system keychain is macOS specific. So, we might be able to omit macOS from the message.